lib/dry/types/default.rb

```diff
@@ -23,7 +24,7 @@ def evaluate
       alias_method :evaluate, :value
 
       # @param [Object, #call] value
-      # @return [Default, Dry::Types::Default::Callable]
+      # @return [Class] `Default` or `Callable`
```

Comment:
I was thinking about such feature in `YARD` some time ago. I would rather see that as a core `YARD` feature or a `YARD` plugin, extending `@return` (and maybe some other tags) syntax, so we have this functionality accessible both programmatically and in human-readable form. 

Then on top of it, it would be good to see corresponding `yardcheck` functionality.

If we could discuss and figure out the syntax that would fit this feature properly then later I could try to build such plugin and expose data needed for `yardcheck`.